src/mono/mono/metadata/class.c

```diff
@@ -870,6 +870,21 @@ inflate_generic_type (MonoImage *image, MonoType *type, MonoGenericContext *cont
 		nt->data.type = inflated;
 		return nt;
 	}
+	case MONO_TYPE_FNPTR: {
+		MonoMethodSignature *in_sig = type->data.method;
+		// quick bail out - if there are no type variables anywhere in the signature,
+		// there's nothing that could get inflated.
+		if (!in_sig->has_type_parameters)
+			return NULL;
+		MonoMethodSignature *new_sig = mono_inflate_generic_signature (in_sig, context, error);
+		if (!new_sig || !is_ok (error))
+			return NULL;
+		if (new_sig == in_sig)
+			return type;
+		MonoType *nt = mono_metadata_type_dup (image, type);
+		nt->data.method = new_sig;
+		return nt;
+	}
```

Comment:
all of the returns here need to be more careful: if `changed` is true, that means the cmods on the `MonoType` got inflated so we have to return either `type` or `nt`.  if `!changed` then we should return `NULL` or `nt`

Reply:
We should be good reusing the control flow of `MONO_TYPE_PTR` case, right?